core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -1189,6 +1189,7 @@ class ConnectionQuotas(config: KafkaConfig, time: Time, metrics: Metrics) extend
   @volatile private var defaultMaxConnectionsPerIp: Int = config.maxConnectionsPerIp
   @volatile private var maxConnectionsPerIpOverrides = config.maxConnectionsPerIpOverrides.map { case (host, count) => (InetAddress.getByName(host), count) }
   @volatile private var brokerMaxConnections = config.maxConnections
+  @volatile private var interBrokerListenerName = config.interBrokerListenerName
```

Comment:
Good point and definitely agree that `config.interBrokerListenerName` could be expensive as we call it several times per accepting a new connection. 

The issue here is that interBrokerListenerName is a dynamic config. So, you will need to update the cached value on changes to that config; similar how we notify ConnectionQuotas about config changes from SocketServer.reconfigure().

Reply:
`interBrokerListenerName` is apparently not a dynamic config.

`DynamicBrokerReconfigurationTest.testAdvertisedListenerUpdate`
...
```
    // Verify updating inter-broker listener
    val props = new Properties
    props.put(KafkaConfig.InterBrokerListenerNameProp, SecureExternal)
    try {
      reconfigureServers(props, perBrokerConfig = true, (KafkaConfig.InterBrokerListenerNameProp, SecureExternal))
      fail("Inter-broker listener cannot be dynamically updated")
   }
```
It does seem like we allow updating the interbroker listener when adding/removing listeners, but that case is covered.
I will try to verify we allow updating the interbroker listener name through the inter broker security protocol, since apparently that's another way to configure the listener.

Reply:
You are right -- I saw that test and thought it was a dynamic config, but the test was verifying that it cannot be updated. I see now that KIP-226 lists that as future work, cool.